pennylane/fourier/reconstruct.py

```diff
@@ -0,0 +1,640 @@
+        nums_frequency (dict[dict]): Numbers of integer frequencies -- and biggest
+            frequency -- per QNode parameter. The keys have to be argument names of ``qnode``
+            and the inner dictionaries have to be mappings from parameter indices to the
+            respective integer number of frequencies. If the QNode frequencies are not contiguous
+            integers, the argument ``spectra`` should be used to save evaluations of ``qnode`` .
+            Takes precedence over ``spectra`` and leads to usage of equidistant shifts.
+        spectra (dict[dict]): Frequency spectra per QNode parameter.
+            The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+            be mappings from parameter indices to the respective frequency spectrum for that
+            parameter. Ignored if ``nums_frequency!=None``.
+        shifts (dict[dict]): Shift angles for the reconstruction per QNode parameter.
+            The keys have to be argument names of ``qnode`` and the inner dictionaries have to
+            be mappings from parameter indices to the respective shift angles to be used for that
+            parameter. For :math:`R` non-zero frequencies, there must be :math:`2R+1` shifts
+            given. Ignored if ``nums_frequency!=None``.
+
+    Returns:
+        function: Function which accepts the same arguments as the QNode.
+        When called, this function will return a dictionary of dictionaries,
+        formatted like ``nums_frequency`` or ``spectra`` ,
+        that contains the univariate reconstructions per QNode parameter.
+
+    For each provided ``id`` in ``ids``, the QNode is restricted to varying the single QNode
+    parameter corresponding to the ``id`` . This univariate function is then reconstructed
+    via a Fourier transform or Dirichlet kernels, depending on the provided input.
+    Either the frequency ``spectra`` of the QNode with respect to its input parameters or
+    the numbers of frequencies, ``nums_frequency`` , per parameter must be provided.
+
+    For quantum-circuit specific details, we refer the reader to
+    `Vidal and Theis (2018) <https://arxiv.org/abs/1812.06323>`__ ,
+    `Vidal and Theis (2020) <https://www.frontiersin.org/articles/10.3389/fphy.2020.00297/full>`__ ,
+    `Schuld, Sweke and Meyer (2021) <https://journals.aps.org/pra/abstract/10.1103/PhysRevA.103.032430>`__ ,
+    and
+    `Wierichs, Izaac, Wang and Lin (2021) <https://arxiv.org/abs/2107.12390>`__ .
+    An introduction to the concept of quantum circuits as Fourier series can also be found in
+    the
+    `Quantum models as Fourier series <https://pennylane.ai/qml/demos/tutorial_expressivity_fourier_series.html>`__
+    and
+    `General parameter-shift rules <https://pennylane.ai/qml/demos/tutorial_general_parshift.html>`__
+    demos as well as the
+    `qml.fourier module docstring <https://pennylane.readthedocs.io/en/stable/code/qml_fourier.html>`__ .
```

Comment:
```suggestion
    :mod:`qml.fourier <pennylane.fourier>` module docstring.
```
This way Sphinx cross links automatically, and it won't direct you to the `stable` docs if you happen to be on `latest`, etc. :)

Reply:
Good catch, thanks!